tasks/benchmark/benches/minifier.rs

```diff
@@ -57,11 +57,15 @@ fn bench_mangler(criterion: &mut Criterion) {
         group.bench_function(id, |b| {
             b.iter_with_setup_wrapper(|runner| {
                 allocator.reset();
-                let program = Parser::new(&allocator, source_text, source_type).parse().program;
-                let semantic =
-                    SemanticBuilder::new().with_scope_tree_child_ids(true).build(&program).semantic;
+                let mut program = Parser::new(&allocator, source_text, source_type).parse().program;
                 runner.run(|| {
-                    let _ = Mangler::new().build_with_semantic(semantic, &program);
+                    let _ = Mangler::new()
+                        .with_options(MangleOptions {
+                            debug: false,
+                            top_level: true,
+                            mangle_private_class_names: true,
+                        })
+                        .build(&allocator, &mut program);
                 });
```

Comment:
this will regress perf a lot as we are now building semantic inside mangler (inside `runner.run`).

I had to make this change because semantic requires a reference to the `Program`, but we need to pass a mutable reference into here for the private class members to mutate the AST.

i'm open to other ideas:
 - we could have dedicted methods for mangling symbols vs private class members
 - something else?